src/components/ChooseDialect.js

```diff
@@ -31,25 +32,33 @@ const ChooseDialect = ({ navigation, dispatch, dialects }) => {
           counter = 0;
         }, 1200);
       } else if (counter === 2) {
-        navigation.navigate("PatternList", { grammars });
+        navigation.navigate("PatternList", { grammars, id });
       }
     } else {
       setSelectedDialectId(id);
       counter = 1;
     }
   };
 
+  const changeBtnText = (id) => {
+    const selectedDialect = dialects.find((dialect) => dialect.id === id);
```

Comment:
since you are doing this twice it might be good to move this to it's own method.